docs/book/src/topics/workload-identity.md

```diff
@@ -0,0 +1,174 @@
+    identity.
+  - The next bullet has link that has steps on creating the federated
+    identity credential. You will need to setup couple of environment
+    variables as mentioned in the sub bullet.
+    - `SERVICE_ACCOUNT_NAMESPACE` : Namespace where the capz-manager pod
+      will run.
+    - `SERVICE_ACCOUNT_NAME` : Name of the capz-manager k8s service account.
+    - `SERVICE_ACCOUNT_ISSUER` : This is the path of the azure storage
+      container which you created in the previous step which is:
+      `"https://${AZURE_STORAGE_ACCOUNT}.blob.core. windows.net/${AZURE_STORAGE_CONTAINER}/"`
+
+  - Create a federated identity credential using the steps outlined [here](https://azure.github.io/azure-workload-identity/docs/topics/federated-identity-credential.html)
+
+- Create a Kind cluster with necessary flags. You can use the following kind 
+  command to do that.
+
+```bash
+cat <<EOF | kind create cluster --name azure-workload-identity --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraMounts:
+    - hostPath: ${SERVICE_ACCOUNT_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.pub
+    - hostPath: ${SERVICE_ACCOUNT_SIGNING_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.key
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+      extraArgs:
+        service-account-issuer: ${SERVICE_ACCOUNT_ISSUER}
+        service-account-key-file: /etc/kubernetes/pki/sa.pub
+        service-account-signing-key-file: /etc/kubernetes/pki/sa.key
+    controllerManager:
+      extraArgs:
+        service-account-private-key-file: /etc/kubernetes/pki/sa.key
+EOF
+```
+
+- Initialize a management cluster using `clusterctl`. Use the below command.
+  If you do not have `clusterctl` installed, then follow this [link](https://cluster-api.sigs.k8s.io/user/quick-start.html#install-clusterctl)
+  to install.
+```bash
+$ clusterctl init --infrastructure azure
+```
+
+### Creating A Workload Cluster
+
+- As mentioned earlier in the doc, cloud provider integration is worked upon
+  and at this stage we need to create an user assigned identity which cloud
+  provider azure can use for authentication. Below are the steps:
+  - [Create](https://learn.microsoft.com/en-gb/azure/active-directory/managed-identities-azure-resources/how-manage-user-assigned-managed-identities?pivots=identity-mi-methods-azp#create-a-user-assigned-managed-identity)
+a user-assigned managed identity in Azure.
+  - [Create a role assignment](https://learn.microsoft.com/en-gb/azure/active-directory/managed-identities-azure-resources/howto-assign-access-portal#use-azure-rbac-to-assign-a-managed-identity-access-to-another-resource)
+to give the identity Contributor access to the Azure subscription where the workload cluster will be created.
+
+- Before we can generate a workload cluster YAML configuration set the
+  following environment variables.
```

Comment:
I think we may need to set `USER_ASSIGNED_IDENTITY_NAME` here as well since it is referenced in the spec.

Reply:
Ah I should've pointed to this line: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3770/files#diff-617e118fb330f6378b8c23d2f2dcb928f31379e840a9b16f2392b59aef139390R143

It is being referenced by the part of the spec we add to the `AzureMachineTemplate` and wasn't specified to be set before in the guide.

Reply:
Oh wait nevermind. You will need to enter them manually anyways so setting the environment variable does not apply in this case. Sorry for the confusion!